src/policy/copyspace.rs

```diff
@@ -89,6 +102,10 @@ impl<VM: VMBinding> Space<VM> for CopySpace<VM> {
     fn release_multiple_pages(&mut self, _start: Address) {
         panic!("copyspace only releases pages enmasse")
     }
+
+    fn set_copy_for_sft_trace(&mut self, semantics: Option<CopySemantics>) {
```

Comment:
We don't have to. Alternatively, we can set both as `Some(DefaultCopy)`. We could even set that in `gc_init()` rather than `prepare()`. However, as we don't copy from `tospace`, setting `tospace`'s copy semantic to `DefaultCopy` would be confusing. I feel it is more conceptually clear to set it in `prepare()` and toggle the semantics for each GC. 

Let me know what you think. I can make the change if you would prefer so.

Reply:
You are right.  It may be confusing if the semantic is `DefaultCopy` while not actually being copied from.

I also anticipate that some spaces may have different copy semantics during different kinds of GC.

Let's keep your code.